File: kotlin-back/src/main/kotlin/com/example/kotlinback/member/refresh/repository/RefreshTokenRedisRepository.java
package com.example.kotlinback.member.refresh.repository;


import com.example.kotlinback.member.refresh.entity.RefreshToken;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Component;

import java.util.Optional;

@Component
public interface RefreshTokenRedisRepository extends CrudRepository<RefreshToken, String> {
    Optional<RefreshToken> findByRefreshToken(String refreshToken);
}

File: kotlin-back/src/main/kotlin/com/example/kotlinback/member/refresh/repository/RefreshTokenRedisRepository.kt
package com.example.kotlinback.member.refresh.repository

import com.example.kotlinback.k.member.refresh.entity.RefreshToken
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Component
import java.util.*

@Component
interface RefreshTokenRedisRepository : CrudRepository<RefreshToken?, String?> {
    fun findByRefreshToken(refreshToken: String?): Optional<RefreshToken?>?
}
